back/web-app/src/users/memoryuser/memory-user.service.ts

```diff
@@ -80,4 +81,12 @@ export class MemoryUserService {
     const user = this.memoryUsers.get(dto.userId);
     user.friends.delete(dto.friendId);
   }
+
+  checkDuplicateNickname(dto: CheckDuplicateNicknameDto) {
+    for (const user of this.memoryUsers.values()) {
+      if (user.nickname == dto.nickname) {
```

Comment:
항상 === strict 비교 연산자를 사용해야 합니다!

Reply:
== 연산자를 사용할 때 어떤 문제가 발생할 수 있을까요?

Reply:
typescript 이고, pipe 로 다 걸러줄 것이라 에러가 날 가능성은 0에 가깝습니다. 하지만 값과 타입을 함께 비교한다는 점에서 우리는 우리 코드를 믿고 돌릴 수 있을 거에요.

우리는 문자열로 저장합니다. 보통 위의 처리를 하지 않는다면 숫자 문자열로 닉네임을 저장할 때 문제가 생길 것 같아요. 문자열 숫자와 정수형 숫자를 같다 하기 때문에 숫자로 들어오면 에러가 날 수 있고, boolean false 값은 '0' 과 같아서 false 로 들어온다 해도 이를 같다 보겠죠?

Reply:
근데 만약 영문자 2 개 이상과 같은 조건을 걸어둔다면 거의 상관 없겠죠? 근데 우린 영문자 및 숫자 가능! 이라고만 할 것이기 때문에 코드 상에서 보험을 들어둬야 할 것 같아요

Reply:
확인했습니다!  === 연산자로 수정하도록 하겠습니다.

참고) 추가로 살펴본 글들 중 도움이 될만한 글을 링크했습니다. ([링크](https://javascript.plainenglish.io/compare-strings-in-javascript-ba372650d4bf#:~:text=The%20strict%20equality%20operator%20is%20more%20reliable%20than%20the%20equality%20operator%20because%20it%20avoids%20type%20coercion.%20Type%20coercion%20can%20lead%20to%20unexpected%20results%20and%20bugs%20in%20your%20code.))

> The strict equality operator is more reliable than the equality operator because it avoids type coercion. Type coercion can lead to unexpected results and bugs in your code.
> 
> 완전 비교 연산자(===)는 강제 타입 변환을 방지하므로 비교 연산자(==)보다 더 안정적입니다.
강제 타입 변환은 코드에서 예기치 않은 결과 및 버그를 유발할 수 있습니다.

Reply:
맞아요! == 는 묵시적 형변환을 해버리기 때문에 위험하다고 하네요. 확인!